calcom/cal.com
# fix: Duplicate Calendar Invites on rescheduling an accepted booking that requires confirmation

## What does this PR do?

Fixes #8444

Reason behind the issue was simply that the code always assumed that rescheduling would move the calendar invite to new time but in case of a booking that requires confirmation it is not true as the booking goes to PENDING state on reschedule.

Added a test as well for the flow
Before Fix  - [Loom](https://www.loom.com/share/d9560369d7f14306a7a1b34f1be5eeca)
After Fix - [Loom](https://www.loom.com/share/c1dc9bccd6cc4ab0abbc8dcb56133810)

## Type of change

## How should this be tested?
- Turn on event approval in settings
- Do a booking as booker
- Accept the booking as organizer
- Reschedule the booking as booker. At this point notice that the calendar event didn't get cancelled, it just moved to new time without confirmation from organizer.
- Confirm the new booking. Now, you would see two calendar invites.


## Mandatory Tasks
